week4/weather/src/@components/todayWeather/todayWeatherInfo.jsx

```diff
@@ -0,0 +1,25 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import WeatherCard from "../common/weatherCard";
+import useWeather from "../../hooks/useWeather";
+import ErrorPage from "../../@pages/errorPage";
+import SkeletonTemplate from "../skeleton/skeletonTemplate";
+
+export default function DetailWeatherInfo() {
+  const { area } = useParams();
+  const { fetchTodayWeatherInfo, isError, todayData, isLoading } = useWeather();
+
+  useEffect(() => {
+    fetchTodayWeatherInfo(area);
+  }, []);
+
+  useEffect(() => {
+    fetchTodayWeatherInfo(area);
+  }, [area]);
+
+  if (isError) {
+    return <ErrorPage />;
+  }
+
+  return <>{isLoading ? <SkeletonTemplate /> : <WeatherCard weatherData={todayData} title={todayData.name} />}</>;
+}
```

Comment:
오오오 여기 리턴으로 isError 조건까지 한번에 처리해줄 수 잇지 아늘까?! 
```suggestion
  return <>{isError? <ErrorPage /> :
      isLoading ? <SkeletonTemplate /> : <WeatherCard weatherData={todayData} title={todayData.name} />}</>;
}
```
요런 늑낌스로.. 소심한 제안

Reply:
이 방법도 죠타! 보통 공식문서들에서 
```
  if (isError) {
    return <ErrorPage />;
  }
```
이렇게 많이들 쓰길래 이렇게 표현해보았는데, 숭희 제안대로 써보는 것도 좋은 것 같아용!!